app/models/batch_connect/session.rb

```diff
@@ -76,9 +76,15 @@ def db_root
       # Find all active session jobs
       # @return [Array<Session>] list of sessions
       def all
-        db_root.children.select(&:file?).map do |f|
-          new.from_json(f.read)
-        end.map do |s|
+        db_root.children.select(&:file?).reject {|p| p.extname == ".bak"}.map do |f|
```

Comment:
I believe that `db_root.children` crashes if `db_root` doesn't exist or isn't a directory or is a directory but isn't executable. I ran into a similar problem with the routers:

https://github.com/OSC/ood-dashboard/commit/8c92bff63d21a15d549566f9436f02513fa76ebc

A guard clause at the top could be as simple as:

```ruby
return [] unless db_root.directory? && db_root.executable? && db_root.readable?
```

Reply:
After protecting against this crash, everything else looks good, assuming it works :-).  No need for me to review again.

Reply:
If it doesn't exist it auto-creates it. Although I do not check if it is a directory that a user has read, write, and execute access to. That can get a little messy to handle all of those situations as `db_root` is called in multiple places and I would need to propagate the errors all the way up to the user in the views.

Reply:
I opened up the issue https://github.com/OSC/ood-dashboard/issues/361 to address the concerns brought up here as it would require its own requirements gathering, design, and PR.